src/cdef.rs

```diff
@@ -526,30 +593,61 @@ pub fn cdef_filter_superblock<T: Pixel, U: Pixel>(
   }
 }
 
-// Input to this process is the Frame rec of reconstructed samples.
-// Output from this process is the Frame rec containing deringed samples.
+#[hawktracer(cdef_filter_tile_group)]
+pub fn cdef_filter_tile_group<T: Pixel>(
+  fi: &FrameInvariants<T>, fs: &mut FrameState<T>, blocks: &mut FrameBlocks,
+) {
+  let tile_size = SB_SIZE * 8;
+  let tile_cols = (fi.config.width + tile_size - 1) / tile_size;
+  let tile_rows = (fi.config.height + tile_size - 1) / tile_size;
+
+  let ti = TilingInfo::from_target_tiles(
+    fi.sequence.sb_size_log2(),
+    fi.config.width,
+    fi.config.height,
+    fi.config.frame_rate(),
+    TilingInfo::tile_log2(1, tile_cols).unwrap(),
+    TilingInfo::tile_log2(1, tile_rows).unwrap(),
+  );
+
+  let in_padded_frame = cdef_padded_frame_copy(&fs.rec);
+  ti.tile_iter_mut(fs, blocks).collect::<Vec<_>>().into_par_iter().for_each(
+    |mut ctx| {
+      cdef_filter_tile(
+        fi,
+        &mut ctx.ts.rec,
+        &ctx.tb.as_const(),
+        &in_padded_frame,
+      );
+    },
+  );
+}
+
+// Input to this process is the array CurrFrame of reconstructed samples and padded input Frame.
+// Output from this process is the array CdefFrame containing deringed samples.
 // The purpose of CDEF is to perform deringing based on the detected direction of blocks.
 // CDEF parameters are stored for each 64 by 64 block of pixels.
 // The CDEF filter is applied on each 8 by 8 block of pixels.
 // Reference: http://av1-spec.argondesign.com/av1-spec/av1-spec.html#cdef-process
-#[hawktracer(cdef_filter_frame)]
-pub fn cdef_filter_frame<T: Pixel>(
-  fi: &FrameInvariants<T>, rec: &mut Frame<T>, blocks: &FrameBlocks,
+#[hawktracer(cdef_filter_tile)]
+pub fn cdef_filter_tile<T: Pixel>(
+  fi: &FrameInvariants<T>, rec: &mut TileMut<'_, T>, tb: &TileBlocks,
+  in_padded_frame: &Frame<u16>,
 ) {
   // Each filter block is 64x64, except right and/or bottom for non-multiple-of-64 sizes.
   // FIXME: 128x128 SB support will break this, we need FilterBlockOffset etc.
-  let fb_width = (rec.planes[0].cfg.width + 63) / 64;
-  let fb_height = (rec.planes[0].cfg.height + 63) / 64;
+  let fb_width = (rec.planes[0].rect().width + 63) / 64;
+  let fb_height = (rec.planes[0].rect().height + 63) / 64;
 
-  // Construct a padded copy of the reconstructed frame.
+  // Construct a part of copy from the padded input frame.
```

Comment:
?  Do you mean:
// Construct a padded copy of part of the input tile?

Reply:
Yes. I will update the comment.